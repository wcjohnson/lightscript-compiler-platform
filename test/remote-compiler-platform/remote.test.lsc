import { InPlacePlatform } from '../..'

test! "load remote compiler platform", -/>
  p = new InPlacePlatform()
  t <- p.createTool({
    source: {
      type: 'remote'
      url: 'https://wcjohnson.github.io/lightscript-compiler/modules/platform/babel-7.1.0.min.js'
    }
  })
  act <- t.apply(null)

  rst <- act.run({
    code: `
      class C {}
    `
  })
  expect(rst.code).toMatchSnapshot()

test! "load toolchain", -/>
  p = new InPlacePlatform()
  t <- p.createTool({
    source: { type: 'toolchain' }
    options: {
      tools: [
        // babel
        {
          source: {
            type: 'remote'
            url: 'https://wcjohnson.github.io/lightscript-compiler/modules/platform/babel-7.1.0.min.js'
          }
        }
        // lsc plugin
        {
          source: {
            type: 'remote'
            url: 'https://wcjohnson.github.io/lightscript-compiler/modules/compiler/4.0.0-alpha.12.min.js'
          }
        }
      ]
    }
  })
  act <- t.apply(null)
  rst <- act.run({
    code: `
      f() -> 1
    `
  })
  expect(rst.code).toMatchSnapshot()

test! "unified toolchain", -/>
  p = new InPlacePlatform()
  t <- p.createTool({
    source: {
      type: 'remote'
      url: 'https://wcjohnson.github.io/lightscript-compiler/modules/registry.js'
    }
  })
  act <- t.apply(null)
  rst <- act.run({
    code: `
      f() -> 1
    `
  })
  expect(rst.code).toMatchSnapshot()
