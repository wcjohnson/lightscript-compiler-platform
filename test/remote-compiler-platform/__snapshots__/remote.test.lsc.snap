// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load remote compiler platform 1`] = `"class C {}"`;

exports[`load toolchain 1`] = `
"function f() {
  return 1;
}"
`;
